### PR TITLE
Regression in to_excel when setting duplicate column names

### LINKED ISSUE #39695: BUG: DataFrame.to_excel() now raises if column parameter contains duplicates
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
import pandas as pd
dft = pd.DataFrame({"A": [0, 1], "B": [10, 11]})
dft.to_excel(r"c:\test\test3.xlsx", columns=["A", "B", "A"])

```

#### Problem description

The example works with pandas 1.1.0, but not with pandas 1.2.1 any more. With 1.2.1 it raises:

```python-traceback
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "c:\test\venv4\lib\site-packages\pandas\core\generic.py", line 2177, in to_excel
    formatter = ExcelFormatter(
  File "c:\test\venv4\lib\site-packages\pandas\io\formats\excel.py", line 470, in __init__
    raise KeyError("Not all names specified in 'columns' are found")
KeyError: "Not all names specified in 'columns' are found"

```

If the column argument doesn't contain duplicates (e.g. `columns=["A", "B"]`) it works also in 1.2.1

In the documentation of .to_excel() I found no information about an intended change in behaviour.

#### Expected Output

I expected it to work like with pandas 1.1.0, producing an excel file with the following content:

image

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 9d598a5e1eee26df95b3910e3f2934890d062caa
python           : 3.9.1.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.18362
machine          : AMD64
processor        : Intel64 Family 6 Model 142 Stepping 10, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : de_DE.cp1252

pandas           : 1.2.1
numpy            : 1.20.0
pytz             : 2021.1
dateutil         : 2.8.1
pip              : 21.0.1
setuptools       : 53.0.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : 3.0.6
pandas_gbq       : None
pyarrow          : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #39695
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

**Comment by gfyoung** [comment]
Changes LGTM.  Restarted CI on failed build since it looks unrelated.

File: doc/source/whatsnew/v1.2.2.rst [hunk 1]
```diff
@@ -27,6 +27,7 @@ Fixed regressions
 - Fixed regression in :meth:`~DataFrame.groupby` or :meth:`~DataFrame.resample` when aggregating an all-NaN or numeric object dtype column (:issue:`39329`)
 - Fixed regression in :meth:`.Rolling.count` where the ``min_periods`` argument would be set to ``0`` after the operation (:issue:`39554`)
 - Fixed regression in :func:`read_excel` that incorrectly raised when the argument ``io`` was a non-path and non-buffer and the ``engine`` argument was specified (:issue:`39528`)
+- Fixed regression in :func:`pandas.to_excel` raising ``KeyError`` when giving duplicate columns with ``columns`` attribute (:issue:`39695`).
```
Comment (simonjayhawkins) [inline doc/source/whatsnew/v1.2.2.rst:30]: move to 1.2.3

Comment (phofl) [reply doc/source/whatsnew/v1.2.2.rst]: Sorry, somehow I am still at 1.2.2 :(

**Comment by jreback** [comment]
thanks @phofl

**Comment by jreback** [comment]
@meeseeksdev backport 1.2.x

**Pull request merged**